Index: Makefile.in
===================================================================
RCS file: /cvs/jikes/jikes/src/Makefile.in,v
retrieving revision 1.14
diff -u -p -r1.14 Makefile.in
--- Makefile.in	2000/01/06 06:46:47	1.14
+++ Makefile.in	2000/01/31 21:24:17
@@ -59,7 +59,7 @@ OBJECTS=jikes.@O@ scanner.@O@ code.@O@ z
         stream.@O@ op.@O@ control.@O@ modifier.@O@ system.@O@ \
         bytecode.@O@ body.@O@ decl.@O@ definite.@O@ dump.@O@ expr.@O@ \
         getclass.@O@ init.@O@ double.@O@ config.@O@ option.@O@ \
-        incrmnt.@O@ case.@O@ tab.@O@ segment.@O@ unparse.@O@
+        incrmnt.@O@ case.@O@ tab.@O@ segment.@O@ unparse.@O@ xml-unparse.@O@
 
 $(EXECUTABLE):	$(OBJECTS)
 	$(LD) $(LDFLAGS)  $(OBJECTS) $(SYSOBJECTS) $(LIBS)
@@ -97,6 +97,14 @@ unparse.@O@:  $(srcdir)/unparse.cpp $(sr
           $(srcdir)/double.h $(srcdir)/symbol.h $(srcdir)/option.h \
           $(srcdir)/depend.h $(srcdir)/access.h 
 	$(CXX) $(CXXFLAGS) $(srcdir)/unparse.cpp 
+
+xml-unparse.@O@:  $(srcdir)/xml-unparse.cpp $(srcdir)/config.h \
+          $(srcdir)/code.h $(srcdir)/case.h $(srcdir)/ast.h \
+          $(srcdir)/stream.h $(srcdir)/javadef.h $(srcdir)/javasym.h \
+          $(srcdir)/tuple.h $(srcdir)/tab.h $(srcdir)/lookup.h $(srcdir)/long.h \
+          $(srcdir)/double.h $(srcdir)/symbol.h $(srcdir)/option.h \
+          $(srcdir)/depend.h $(srcdir)/access.h 
+	$(CXX) $(CXXFLAGS) $(srcdir)/xml-unparse.cpp 
 
 body.@O@: $(srcdir)/body.cpp $(srcdir)/config.h $(srcdir)/semantic.h \
           $(srcdir)/ast.h $(srcdir)/stream.h $(srcdir)/javadef.h \
Index: ast.h
===================================================================
RCS file: /cvs/jikes/jikes/src/ast.h,v
retrieving revision 1.18
diff -u -p -r1.18 ast.h
--- ast.h	1999/10/19 23:13:34	1.18
+++ ast.h	2000/01/31 21:24:27
@@ -393,6 +393,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     //
@@ -817,6 +818,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -853,6 +855,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -884,6 +887,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -933,6 +937,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -975,6 +980,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1005,6 +1011,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1036,6 +1043,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1088,8 +1096,11 @@ public:
 
 #ifdef TEST
     virtual void Print(LexStream &);
-    virtual void Unparse(LexStream &, char * directory); // special form
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
+
+    virtual void Unparse(LexStream &, char * directory); // special form
+    virtual void XMLUnparse(LexStream &, char * directory); // special form
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1146,6 +1157,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1175,6 +1187,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1291,6 +1304,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1367,6 +1381,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1414,6 +1429,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1455,6 +1471,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1492,6 +1509,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1549,6 +1567,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1593,6 +1612,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1645,6 +1665,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1710,6 +1731,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1742,6 +1764,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1801,6 +1824,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1865,6 +1889,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1924,6 +1949,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -1974,6 +2000,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2098,6 +2125,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2149,6 +2177,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2211,6 +2240,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2251,6 +2281,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2288,6 +2319,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2329,6 +2361,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2359,6 +2392,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2411,6 +2445,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2473,6 +2508,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2511,6 +2547,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2551,6 +2588,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2610,6 +2648,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2649,6 +2688,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2687,6 +2727,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2725,6 +2766,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2763,6 +2805,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2801,6 +2844,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2837,6 +2881,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2867,6 +2912,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2911,6 +2957,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -2980,6 +3027,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3011,6 +3059,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3042,6 +3091,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3072,6 +3122,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3102,6 +3153,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3132,6 +3184,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3162,6 +3215,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3192,6 +3246,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3222,6 +3277,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3252,6 +3308,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3283,6 +3340,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3316,6 +3374,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3347,6 +3406,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3412,6 +3472,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3446,6 +3507,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3497,6 +3559,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3574,6 +3637,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3632,6 +3696,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3666,6 +3731,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3724,6 +3790,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3782,6 +3849,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3833,6 +3901,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3911,6 +3980,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -3946,6 +4016,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
@@ -4029,6 +4100,7 @@ public:
 #ifdef TEST
     virtual void Print(LexStream &);
     virtual void Unparse(Ostream &, LexStream &);
+    virtual void XMLUnparse(Ostream &, LexStream &);
 #endif
 
     virtual Ast *Clone(StoragePool *);
Index: control.cpp
===================================================================
RCS file: /cvs/jikes/jikes/src/control.cpp,v
retrieving revision 1.29
diff -u -p -r1.29 control.cpp
--- control.cpp	2000/01/18 18:45:04	1.29
+++ control.cpp	2000/01/31 21:25:05
@@ -1363,6 +1363,17 @@ void Control::CleanUp(FileSymbol *file_s
         }
         if (option.debug_dump_ast)
             sem -> compilation_unit -> Print(*sem -> lex_stream);
+        if (option.debug_xml_unparse_ast)
+        {
+            if (option.debug_unparse_ast_debug)
+              {
+                // which of these is correct?
+                sem -> compilation_unit -> debug_unparse = true;
+                Ast::debug_unparse = true;
+              }
+            sem -> compilation_unit -> XMLUnparse(*sem -> lex_stream,
+                                                  "xml-unparsed/");
+        }
         if (option.debug_unparse_ast)
         {
             if (option.debug_unparse_ast_debug)
Index: jikes.cpp
===================================================================
RCS file: /cvs/jikes/jikes/src/jikes.cpp,v
retrieving revision 1.65
diff -u -p -r1.65 jikes.cpp
--- jikes.cpp	2000/01/11 02:50:15	1.65
+++ jikes.cpp	2000/01/31 21:25:06
@@ -77,6 +77,8 @@ int main(int argc, char *argv[])
         "+O numbytes        call no-op op_trap() for bytecodes of the given length\n"
         "+u                 unparse AST; produces Java code for the AST\n"
         "+ud                unparse AST, with extra debugging information\n"
+        "+ux                unparse AST into JavaML XML representation\n"
+        "+udx               unparse AST into JavaML, with extra debugging info\n"
 #endif
         "\nVersion 1.11 (10 Jan 2000)\n"
         "Originally written by Philippe Charles and David Shields \n"
Index: option.cpp
===================================================================
RCS file: /cvs/jikes/jikes/src/option.cpp,v
retrieving revision 1.31
diff -u -p -r1.31 option.cpp
--- option.cpp	2000/01/14 20:34:23	1.31
+++ option.cpp	2000/01/31 21:25:07
@@ -321,6 +321,16 @@ Option::Option(ArgumentExpander &argumen
                 debug_unparse_ast = true;
                 debug_unparse_ast_debug = true;
             }
+            else if (strcmp(arguments.argv[i], "+ux") == 0)
+            {
+                debug_xml_unparse_ast = true;
+            }
+            else if (strcmp(arguments.argv[i], "+udx") == 0)
+            {
+                debug_unparse_ast_debug = true;
+                debug_xml_unparse_ast = true;
+            }
+
 #ifdef EBCDIC
             else if (strcmp(arguments.argv[i], "+ASCII") == 0)
                      ascii = true;
Index: option.h
===================================================================
RCS file: /cvs/jikes/jikes/src/option.h,v
retrieving revision 1.18
diff -u -p -r1.18 option.h
--- option.h	2000/01/07 21:23:58	1.18
+++ option.h	2000/01/31 21:25:07
@@ -138,6 +138,7 @@ public:
     bool debug_dump_lex,
          debug_dump_ast,
          debug_unparse_ast,
+         debug_xml_unparse_ast,
          debug_unparse_ast_debug,
          debug_dump_class,
          nocleanup,
